conda/common/iterators.py

```diff
@@ -8,12 +8,17 @@
 import itertools
 import collections
 
+
 def groupby_to_dict(keyfunc, sequence):
     """
     toolz-style groupby, returns a dictionary of { key: [group] } instead of
     iterators.
     """
-    result = collections.defaultdict(lambda: [])
+    result = collections.defaultdict(list)
     for key, group in itertools.groupby(sequence, keyfunc):
         result[key].extend(group)
     return dict(result)
+
+
+def concat(seqs):
+    return itertools.chain.from_iterable(seqs)
```

Comment:
Doesn't belong in this PR

Reply:
Would you mind if I combine the two related PR's? Both edit subdir_data.py, otherwise one PR will cause merge conflicts with the second PR.

Reply:
https://github.com/conda/conda/pull/12164/files

Reply:
Combined